### PR TITLE
fix: ignore auth for swagger endpoint

### PR DESCRIPTION
**Pull request by babakks**
This PR disables authentication for the `/swagger.json` endpoint.

Fixes CSS-10917

### CODE REVIEW

File: v1/core_test.go [hunk 1]
```diff
@@ -79,3 +82,37 @@ func TestHandlerWorksWithChiMux(t *testing.T) {
...
+// middleware.
+func TestSwaggerJsonIgnoresAuthentication(t *testing.T) {
+	c := qt.New(t)
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+
+	// A mock authenticator that always returns an authentication failure error,
+	// but it expects no calls. So, the test will fail if the mock struct method(s)
+	// get called.
+	authenticator := interfaces.NewMockAuthenticator(ctrl)
```
Comment (SimoneDutto) [inline v1/core_test.go:97]: question: where is this behavior described in the comment set?

File: v1/authentication_test.go [hunk 2]
```diff
@@ -213,3 +214,50 @@ func TestContextualAuthenticatedIdentity_MiddlewareAndContext(t *testing.T) {
...
+
+func TestAuthenticationIgnoresCertainEndpoints(t *testing.T) {
+	c := qt.New(t)
+
+	// Slice of endpoints that should not require authentication.
+	endpoints := []struct {
+		method string
+		path   string
+	}{
+		{
```
Comment (SimoneDutto) [inline v1/authentication_test.go:226]: just asking, but I imagine you have other parts in the code where you test auth is working.

**Review by SimoneDutto — APPROVED** [review]
lgtm

File: v1/core_test.go (hunk 1 above)
Comment (babakks) [reply v1/core_test.go:97]: It's the `gomock` behaviour. When you create a new mock instance, you should then head out to set your expectations (e.g., by calling `authentiator.EXPECT(). ...`). If you don't set any expectation it means no calls are expected.

File: v1/authentication_test.go (hunk 2 above)
Comment (babakks) [reply v1/authentication_test.go:226]: We test it in two points actually:
- One in the `core_test.go` does a full-handler test (from the HTTP server, down to the actual handler). This is the desired behavior we need to test; kind of like an integration test (between the components of the library).
- Another in the `authentication_test.go` which is more like a unit test for the authentication middleware, to verify that it doesn't invoke the user-populated authenticator middleware if it's `/swagger.json`.

Comment (SimoneDutto) [reply v1/authentication_test.go:226]: ty

File: v1/core_test.go (hunk 1 above)
Comment (SimoneDutto) [reply v1/core_test.go:97]: ty

**Review by pkulik0 — APPROVED** [review]
nice. back to square one with pass through for swagger.json 😉 size S

// edit: ah not square one, had the comment window open for both rebac auth PRs and thought I was writing the one for jimm because of almost the same name lol. Nonetheless, good changes

**Pull request merged**